File: C#/Step_10_StringManipulation_Basic.cs
using System;

class Program
{
    static void Main()
    {

        string str1 = "Hello";
        string str2 = "World";

        string str3 = str1 + " " + str2;

        Console.WriteLine(str3); //We can add two strings and store them in another

        str1 = str1 + " " + str2;

        Console.WriteLine(str1); //We can also add another string to a string and update it

        string multi = @"This is a
        multiline string";

        Console.WriteLine(multi);

        string lower = "rafy";
        lower = lower.ToUpper(); //We can make it uppercase
        Console.WriteLine(lower);

        //Can you guess how we can make it lowercase?

        string name = "Rafy";

        string booL = Convert.ToString(name.Contains("a")); //it checks if the strin contains a certain substring in it. [Equivalent to "a in name" in python]

        Console.WriteLine(booL);

        booL = Convert.ToString(name.StartsWith("b")); //Checks if it starts with a certain string

        Console.WriteLine(booL);
        //Can you guess the syntax of ending?

        int index_ = name.IndexOf("a"); //Tracks the index number of a certain part
        Console.WriteLine(index_);

        index_ = name.IndexOf("af");
        Console.WriteLine(index_); //Even if you are putting more than one characters, it will return the index of the first character.

        //Then why does this has this option fetch the index number of more than one character?
        //Because you might need to find in which index a certain word starts

        string sentence_ = "Hello there, are you thirsty? is he thirsty?";
        //In the string above the substrin 't' is in three words.
        //So if you want to find out at which index the word thirsty starts, you will need the option to search by using more than one character.
        //Let's check it
        index_ = sentence_.IndexOf("thirsty"); //However it always picks the first appearence
        Console.WriteLine(index_);
        //Now can you check what happens if you search something which is not there?
        sentence_ = sentence_.Replace("thirsty", "angry"); //Replaces the first substring with the second string
        Console.WriteLine(sentence_);

        string substrin = sentence_.Substring(1,3); // Cuts a String. Equivalent to python's string[start:end] but the ending in inclusive. Meaning it will also take the ending index.
        Console.WriteLine(substrin);




    } 
}

File: C Program/Step_10_StringManipulation_Basic.c
#include <stdio.h>
#include <string.h> //Without this, we can't use strlen(). These are libraries like in python
#include <strings.h> //To compare strings avoiding case

int main(){

    char str1[] = "Hello";//The length is not declared here 

    printf("The length of the string 'str1' is: %d\n", strlen(str1));
    char str2[10] = "World";//This string can contain 9 characters at max. The 10th one is for \0

    char name[50]; //We just declared here that in some point, I might use a string named "name" that can contain at max 49 Characters.
    printf("Enter your first name: ");
    scanf("%s", name); //Used "scanf" to take an input, I said C that, "Hey, take an input which is a string[%s]"
                       //I also let the program know that the string name is "name" where the input will get stored;
                       //But there is a catch, you can only input something that does not have spaces. Like a full name.
                       //Try it and check what happens.
                       //It there any solution for that?
                       //Of course, anything that we think can be done, can be done.
    
    printf("The first name is: %s\n", name);


    //=======================================================This part is explained better in the Advance files. [Step_10A_StringManipulation_Advance.c]
    // char full_name[50];
    // printf("Input your full name: ");

    // // Clear the input buffer
    // int ch;
    // while ((ch = getchar()) != '\n') {}

    // // while ((ch = getchar()) != '\n' && ch != EOF) {} \\ \\ This is a better solution.


    // fgets(full_name, sizeof(full_name), stdin);

    // printf("The full name is: %s", full_name);
    //========================================================[It's okay to skip it now]


    //Now let's get back to basics

    char haha[] = "haha";

    char hihi[10];

    strcpy(hihi, haha); //This syntax says copy haha, into hihi. Equivalent to python's string.copy()

    printf("%s\n", hihi);

    char hehe[] = "hehe";
    char hoho[] = "hoho";

    strcat(hehe,hoho); // In python, it is 'hyhy = hehe + hoho'

    printf("%s\n", hehe);

    strcat(hoho,hehe);

    printf("%s\n", hoho); //Can you explain why it printed hohohehehoho and not hohohehe

    char comp_1[] = "Rafy";
    char comp_2[] = "Radar";

    if (strcmp(comp_1, comp_2) == 0){
        printf("Same string");
    }
    else if((strcmp(comp_1, comp_2)<0)){
        
        printf("%s is less than %s", comp_1, comp_2);
    }
    else if((strcmp(comp_1, comp_2)>0)){
        
        printf("%s is greater than %s", comp_1, comp_2); //The inequality is compared lexicographically. More like a dictionary.
    }
    else{
        printf("Not same string");
    }
    printf("\n");
    //Now, if we want to compare to strings by their meaning like Word and word.
    //These two words mean the same, but one having a capital char and one not can cause it to return otherwise.

    char comp_3[] = "Word";
    char comp_4[] = "worD";
    
    if(strcasecmp(comp_3,comp_4) == 0){
        printf("Same\n");
    }
    else{
        printf("Not Same\n");
    }

    
    char sentence_[] = "Hello, I am under the water, here too much raining wooooo";
    if (strstr(sentence_, "water")!=NULL){
        printf("Found\n");
    }
    else{
        printf("Not Found\n");
    }


    

    return 0;
}